investigations/Possible_Instrument_View_Improvements.md

```diff
@@ -0,0 +1,48 @@
+- Why is `PickID` stored as some strange encoded colour? Couldn't some kind of map be built using the `DetectorID` directly so that on mouse over of a `ProjectionSurface` one could immediately retrieve the `DetectorID`? - ***"The pick color is a convenient way of addressing detectors using only 3 bytes (RGB ingoring Alpha) in OpenGL space. It makes it easier to handle mouse over on the visualized geometry. Detector IDs are sufficiently large to exceed the three bytes."***
+- Will workspace indices be directly accessible from `DetectorInfo`? ***"Not necessarily, however, a map between detector indices/IDs can be built and used if necessary. Also, there is SpectrumInfo which maps spectrum index to detector indices in the SpectrumDefinition.***
+
+### Recommendations
+
+#### Calls to ExperimentInfo::getInstrument()
+
+It seems as though calls to `ExperimentInfo::getInstrument()` are remnants of a legacy system where instrument geometry and associated parameter maps were always calculated on the fly. Since geometry is stored (in both Instrument 1.0 and 2.0) and the workspace observer mechanism allows for response to workspace updates, this is not at all necessary. These should be removed in favour of storing the `ComponentInfo/DetectoInfo`
+
+#### Actors 
+
+The actors seem to be a middle layer which wrap instrument, component and object operations. These layers interact with workspaces, parameter maps, store geometry and store colour (detector pixel intensity and pick colours). Roman believes with the new Instrument 2.0 layers `ComponentInfo/DetectorInfo` these may be redundant and in many cases may only act to slow things down. An example is in the constructor of `CompAssemblyActor` [here](https://github.com/mantidproject/mantid/blob/master/qt/widgets/instrumentview/src/CompAssemblyActor.cpp#L41) where there are several dynamic casts in a tight loop in order to check component type. Removing the actors would not only get rid of unnecessary overheads, but will create a more lightweight Instrument View widget solely concerned with displaying geometry and associated data. Instrument 2.0 layers already effectively and efficiently store all geometry in memory, even specifics like whether or not we are dealing with rectangular detectors.
+
+#### Faster displays
+
+Roman suggested using OpenGL display lists to speed up rendering although it seems like display lists are already extensively used for drawing the "actors" which store colour information. See [here](https://github.com/mantidproject/mantid/blob/master/qt/widgets/instrumentview/src/GLActorCollection.cpp#L39)
+
```

Comment:
This is a good point. I cannot find anything which suggests detector ids are in any way optimal. However, I will investigate this more when I come up with the sketch design.